src/macros.rs

```diff
@@ -343,6 +356,29 @@ macro_rules! declare_bitpacker {
             }
         }
 
+        struct StrictDeltaIntegrate {
+            current: DataType,
+            output_ptr: *mut DataType,
+        }
+
+        impl StrictDeltaIntegrate {
+            unsafe fn new(initial: u32, output_ptr: *mut DataType) -> StrictDeltaIntegrate {
+                StrictDeltaIntegrate {
+                    current: set1(initial as i32),
+                    output_ptr,
+                }
+            }
+        }
+
+        impl Sink for StrictDeltaIntegrate {
+            #[inline]
+            unsafe fn process(&mut self, delta: DataType) {
+                self.current = integrate_delta(self.current, add(delta, set1(1)));
```

Comment:
Minor comment. Non trivial question.

Should we check if the perf is better if we do 

`add(integrate_delta(self.current, delta), staircase_register)`

where stair_case register contains `[1,2,3,4...]`

I suspect the latter might be a tad faster due to dependencies, but it depends on integrate_delta.

Reply:
I gave it a try with this commit https://github.com/quickwit-oss/bitpacking/commit/162356b2eff7bc835305c64571802ea0ccfc5908, and the changes were withing noise threshold.